Cataclysm/Warrior.lua

```diff
@@ -1130,288 +1158,120 @@ spec:RegisterAbilities( {
             if talent.juggernaut.enabled then return target.minR > 7, "target must be outside your deadzone" end
             return (talent.warbringer.enabled or not combat) and target.minR > 7, "cannot be in combat; target must be outside your deadzone"
         end,
-
         handler = function( rank )
             setDistance( 7 )
             if not target.is_boss then applyDebuff( "target", "charge_stun" ) end
             applyBuff( "juggernaut" )
         end,
     },
 
-    -- On next attack...
-    cleave = {
-        id = 47520,
-        cast = 0,
-        cooldown = 0,
-        gcd = "off",
+        -- On next attack...
+        cleave = {
+            id = 47520,
+            cast = 0,
+            cooldown = 0,
+            gcd = 3,
 
-        spend = function() return 20 - talent.focused_rage.rank end,
-        spendType = "rage",
+            spend = function() return 20 - talent.focused_rage.rank end,
+            spendType = "rage",
 
-        startsCombat = true,
-        texture = 132338,
+            startsCombat = true,
+            texture = 132338,
 
-        nobuff = "cleave",
+            nobuff = "cleave",
 
-        usable = function()
-            return (not buff.heroic_strike.up) and (not buff.cleave.up)
-        end,
+            usable = function()
+                local currentTime = GetTime()
+                local effective_GCD = inner_rage_active and shared_GCD / 2 or shared_GCD
+                return (currentTime - last_heroic_strike >= effective_GCD) and (currentTime - last_cleave >= effective_GCD)
```

Comment:
Talk me through these calculations and the goal for Cleave and Heroic Strike?

Reply:
function that checks if a certain action is usable based on  (GCD) and the last time the actions heroic_strike and cleave were used.

Reply:
I think this might be encroaching on work the addon is already doing behind the scenes.  I guess I need to know the specific problem being solved by this code.  At the very least, this isn't compatible with the addon's timing system or looking at next steps beyond the current moment.

Reply:
Wouldn't !IsCurrentSpell("Cleave") or !IsCurrentSpell("Heroic Strike") work? as it returns true if they aren't queued?